include/iceoryx_dds/dds/fast_context.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef FAST_CONTEXT_HPP
+#define FAST_CONTEXT_HPP
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+
+namespace iox
+{
+namespace dds
+{
+///
+/// @brief The FastContext manages fast configurations and singleton artifacts shared throughout an application.
+///
+class FastContext
+{
+  public:
+    ///
+    /// @brief getParticipant Get the DDS Domain Participant for the current runtime.
+    /// @return The DDS Domain Participant.
+    ///
+    static eprosima::fastdds::dds::DomainParticipant* getParticipant();
```

Comment:
Please don't make everything static, you the scott meyers singleton approach. The idea is, that you have a normal object, in this case:

```cxx
class FastContext
{
  public:
    eprosima::fastdds::dds::DomainParticipant* getParticipant();

    // add this here:
    static FastContext& get_instance() {
        static FastContext context;
        return context;
    } 

  private:
    void initialize();

    eprosima::fastdds::dds::DomainParticipant* m_participant;
    eprosima::fastdds::dds::TypeSupport m_type;
};
```

The reason is, that C++ itself has the static initialization order fiasco problem and making a lot of member variables static does not help if you ever run into this problem.
Another advantage, you can use this object now also as a non-singleton which is essential for testing!